### DISCUSSION TITLE
Running Belay from Micropython itself.

### DISCUSSION THREAD

**Discussion by rkompass** [BrianPugh/belay#143] [General]
Hi Brian,

what a nice library. 
It may help in my MP developments in the following way:

I was starting to write a generator for certain patterns of quadrature coding, including simulated switch bounces. So that quadrature encoders (there are many versions of those, some better/faster some worse/slower) may be tested and compared and their performance / limit frequencies be measured.
I would run the generator on the RP2 Pico using it's PIO.
Now to test the performance of the quadrature encoders on other platforms these have to be set up and their results, i.e. rotation counts, have to be compared against the generated true values.
For that Belay seems ideal. **Provided it may run on the RP2 Pico in Micropython itself.**
Is that possible? Did you try that already?

What kind of connection would be necessary? Would serial port suffice? Without handshaking?
Would perhaps even a 1-wire solution work, if set up properly?
Thx

**Comment by BrianPugh** [comment]
Hi rkompass!

Apologies, but I'm a little confused on what the exact question is. Belay's main strengths is that it makes it simpler to develop for micropython on a desktop, especially in cases where a desktop python script needs to interact with real world devices, such as a quadrature encoder. You could certainly use Belay for your application, but it's benefits certainly depend on how you want to setup your project.

**Reply by rkompass** [reply]
O.k., my question was too long/complicated. Shorter:
Belay runs on the Desktop (CPython) and controls a smaller controller with Micropython on it.
Now, is it possible to
run Belay on a small controller (in Micropython) and control another small controller with Micropython on it?

**Reply by BrianPugh** [reply]
gotcha 😄 . No, Belay cannot run on a microcontroller on micropython; it uses a few libraries that are not available on micropython, and many features available with cpython that are unavailable with micropython.
